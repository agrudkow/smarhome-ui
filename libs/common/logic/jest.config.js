module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
