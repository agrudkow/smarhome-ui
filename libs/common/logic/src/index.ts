export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
