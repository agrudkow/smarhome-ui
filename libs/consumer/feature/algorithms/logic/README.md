# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
