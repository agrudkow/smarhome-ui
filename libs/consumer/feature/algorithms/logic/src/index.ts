export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
