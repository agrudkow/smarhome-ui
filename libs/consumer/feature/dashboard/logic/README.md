# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
