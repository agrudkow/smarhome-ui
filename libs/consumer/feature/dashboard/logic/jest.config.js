module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
