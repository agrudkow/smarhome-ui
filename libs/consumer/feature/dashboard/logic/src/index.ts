export * from './lib/use-billings';
