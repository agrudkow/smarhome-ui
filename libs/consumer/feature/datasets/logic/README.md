# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
