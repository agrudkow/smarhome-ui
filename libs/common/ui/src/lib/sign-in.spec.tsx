import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { customRender } from '@smarthome/common/logic';
import SignIn from './sign-in';

describe(' SignIn', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <Router>
        <SignIn clientId="" navlinkRoute="" />
      </Router>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <Router>
        <SignIn clientId="" navlinkRoute="" />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
