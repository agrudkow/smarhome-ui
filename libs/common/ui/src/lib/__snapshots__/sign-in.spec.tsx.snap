// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` SignIn should match snapshot 1`] = `
.c1 {
  padding: 5px 15px;
  margin: 15px 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 300px;
  width: 300px;
  max-height: 300px;
  max-width: 300px;
  background-color: #102128;
  opacity: 1;
  border-radius: 6px;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.c0 {
  height: 100vh;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E3E2E1;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  margin: 30px;
  padding: 1px;
  border: 2px solid #1BBDCE;
  border-radius: 6px;
}

.c4 .google-sign-in-button {
  box-shadow: none !important;
  font-size: inherit !important;
  font-weight: inherit !important;
  padding-right: 10px !important;
}

.c3 {
  color: #E3E2E1;
  white-space: nowrap;
  margin: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 30px;
  line-height: 30px;
}

.c5 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  padding: 0;
  margin: 0;
  color: white;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (max-width:769px) {
  .c1 {
    margin: 5px 0;
  }
}

@media (max-width:480px) {
  .c1 {
    padding: 5px 5px;
  }
}

@media (min-width:769px) {
  .c5 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 16px;
    line-height: 24px;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        height="300"
        width="300"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            Smarthome
          </div>
          <div
            class="c4"
          >
            <button
              class="google-sign-in-button"
              disabled=""
              style="background-color: rgb(255, 255, 255); display: inline-flex; align-items: center; color: rgba(0, 0, 0, 0.54); box-shadow: 0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24); padding: 0px; border-radius: 2px; border: 1px solid transparent; font-size: 14px; font-weight: 500; font-family: Roboto, sans-serif; opacity: 0.6;"
              type="button"
            >
              <div
                style="margin-right: 10px; background: rgb(255, 255, 255); padding: 10px; border-radius: 2px;"
              >
                <svg
                  height="18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#000"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                      fill="#EA4335"
                    />
                    <path
                      d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                      fill="#4285F4"
                    />
                    <path
                      d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                      fill="#FBBC05"
                    />
                    <path
                      d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                      fill="#34A853"
                    />
                    <path
                      d="M0 0h18v18H0z"
                      fill="none"
                    />
                  </g>
                </svg>
              </div>
              <span
                style="font-weight: 500; padding: 10px 10px 10px 0px;"
              >
                Sign in with Google
              </span>
            </button>
          </div>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <p
              class="c5"
            >
              Create a new account
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
